trinity/rpc/ipc.py

```diff
@@ -1,33 +1,46 @@
 import asyncio
 import json
 import logging
-import os
 
 from cytoolz import curry
 
-from trinity.rpc.main import RPCServer
+from p2p.cancel_token import (
+    CancelToken,
+    wait_with_token,
+)
+from p2p.exceptions import (
+    OperationCancelled,
+)
 
 
 MAXIMUM_REQUEST_BYTES = 10000
 
 
 @curry
-async def connection_handler(execute_rpc, reader, writer):
+async def connection_handler(execute_rpc, cancel_token, reader, writer):
     '''
     Catch fatal errors, log them, and close the connection
     '''
     logger = logging.getLogger('trinity.rpc.ipc')
+
     try:
-        await connection_loop(execute_rpc, reader, writer, logger)
+        await wait_with_token(
```

Comment:
Is this enough? is this in the right place?

Reply:
Nope, you need to pass the token all the way down until the function/method that calls the asyncio APIs, and there you use `wait_with_token()`. In this case that seems to be `connection_loop()` and `write_error()`